root/static/scripts/edit/URLCleanup.js

```diff
@@ -4300,6 +4303,36 @@ const CLEANUPS: CleanupEntries = {
       return url;
     },
   },
+  'vimeoondemand': {
+    match: [new RegExp(
+      '^(https?://)?([^/]+\\.)?vimeo\\.com/(?:ondemand|store/ondemand)',
+      'i',
+    )],
+    restrict: [LINK_TYPES.downloadpurchase, LINK_TYPES.streamingpaid],
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:[^\/]+\.)?vimeo\.com\/ondemand\/([^\/?#]+)(?:.*)$/, 'https://vimeo.com/ondemand/$1');
+      return url;
+    },
+    validate: function (url) {
+      const isStoreLink = /^https:\/\/vimeo\.com\/store\/ondemand/.test(url);
+      if (isStoreLink) {
+        return {
+          error: exp.l(
+            `Please link to the main “{video_url_pattern}” page
+             rather than this store link.`,
+            {
+              video_url_pattern: (
+                <span className="url-quote">{'/ondemand'}</span>
+              ),
+            },
+          ),
+          result: false,
+          target: ERROR_TARGETS.URL,
+        };
+      }
+      return {result: true};
```

Comment:
Further validation should happen here to prevent linking to either `artist` or `label`.

Reply:
I guess that might make sense, yes :) Not sure whether we should just restrict it to recording only?

Reply:
Or still allow release? Discussion is still going on at https://github.com/metabrainz/musicbrainz-server/pull/2261#discussion_r711110828

Reply:
The outcome of https://github.com/metabrainz/musicbrainz-server/pull/2261#discussion_r713729787 is “for now allow video links for releases and then consolidate whatever happens later”.